lib/gs-plugin-loader.c

```diff
@@ -3274,6 +3276,9 @@ gs_plugin_loader_init (GsPluginLoader *plugin_loader)
 	for (i = 0; projects[i] != NULL; i++)
 		g_debug ("compatible-project: %s", projects[i]);
 	priv->compatible_projects = projects;
+
+	priv->global_plugin_statuses = g_hash_table_new (g_str_hash,
```

Comment:
there should be a corresponding disposal of `priv->global_plugin_statuses` in `gs_plugin_loader_dispose()`. And please make sure that you free the keys (if necessary) as well as the structure

Reply:
Is this different from doing it in gs_plugin_loader_finalize? How do i determine which one's appropriate. ie between gs_plugin_loader_dispose and gs_plugin_loader_finalize.

Reply:
Please see the discussion here on dispose vs. finalize in GObject: https://mail.gnome.org/archives/gtk-devel-list/2004-July/msg00151.html

Reply:
This might also be useful; in short, `dispose()` can be called multiple times whereas `finalize()` only happens when the memory will actually be freed: https://developer.gnome.org/gobject/stable/howto-gobject-destruction.html